pkgs/games/quakespasm/default.nix

```diff
@@ -1,27 +1,78 @@
-{ lib, stdenv, SDL, fetchurl, gzip, libvorbis, libmad, copyDesktopItems, makeDesktopItem }:
+{ lib, stdenv, SDL, SDL2, fetchurl, gzip, libvorbis, libmad
+, Cocoa, CoreAudio, CoreFoundation, IOKit, OpenGL
+, copyDesktopItems, makeDesktopItem
+, useSDL2 ? stdenv.isDarwin # TODO: CoreAudio fails to initialize with SDL 1.x for some reason.
+}@args:
 
+let
+  macFrameworkNames = [
+    "Cocoa" "CoreAudio" "IOKit" "OpenGL"
+  ] ++ lib.optionals useSDL2 [
+    "CoreFoundation"
+  ];
+  macFrameworks = map (f: args.${f}) macFrameworkNames;
+in
 stdenv.mkDerivation rec {
   pname = "quakespasm";
-  majorVersion = "0.93";
-  version = "${majorVersion}.2";
+  majorVersion = "0.94";
+  version = "${majorVersion}.1";
 
   src = fetchurl {
-    url = "mirror://sourceforge/quakespasm/quakespasm-${version}.tgz";
-    sha256 = "0qm0j5drybvvq8xadfyppkpk3rxqsxbywzm6iwsjwdf0iia3gss5";
+    url = "mirror://sourceforge/quakespasm/quakespasm-${version}.tar.gz";
+    sha256 = "19grpvsk3ikjypx9j0gpfsx5wanrqxkgf8dwl9h6ab5c8wwmjcjp";
   };
 
   sourceRoot = "${pname}-${version}/Quake";
 
+  patches = lib.optionals stdenv.isDarwin [
+    ./no-bundled-dependencies.patch
```

Comment:
Could use better name. This reworks the Makefile.darwin to be more like generic Unix Makefile is (using system libraries instead of assuming libraries are inside app bundle)